dace/sdfg/dealias.py

```diff
@@ -0,0 +1,153 @@
+    """
+    recurse_in: Set[dace.nodes.NestedSDFG] = set()
+
+    for state in sdfg.all_states():
+        for node in state.nodes():
+            if isinstance(node, dace.nodes.NestedSDFG):
+                recurse_in.add(node)
+
+                in_edges = state.in_edges(node)
+                out_edges = state.out_edges(node)
+
+                # Gather all replacements we need
+                # E.g.                      A[::] -> tmpxceX (NestedSDFG)
+                # Needs to be replaced with A[::] -> A_slice (NestedSDFG)
+                # A_slice is chosen if the subset is different than the complete shape A
+                # Otherwise A is chosen
+                # Also consider the case where A[i] -> tmp1 (NestedSDFG)
+                #                              A[j] -> tmp2
+                # In this case we need not map them to A twice but to A_slice1, A_slice2
+                input_repldict = dict()
+                output_repldict = dict()
+                for in_edge in in_edges:
+                    # Skip "__return"
+                    if in_edge.data is not None and in_edge.data.data == "__return":
+                        continue
+                    if in_edge.data is not None and in_edge.data.data != in_edge.dst_conn:
+                        new_connector = _get_new_connector_name(in_edge, input_repldict, output_repldict, state,
+                                                                node.sdfg)
+                        input_repldict[in_edge.dst_conn] = new_connector
+
+                for out_edge in out_edges:
+                    if out_edge.data is not None and out_edge.data.data == "__return":
+                        continue
+                    if out_edge.data is not None and out_edge.data.data != out_edge.src_conn:
+                        new_connector = _get_new_connector_name(out_edge, output_repldict, input_repldict, state,
+                                                                node.sdfg)
+                        output_repldict[out_edge.src_conn] = new_connector
+
+                # Replace connectors rm tmpxceX connector with A
+                for dst_name in set(input_repldict.keys()):
+                    rmed = node.remove_in_connector(dst_name)
+                    assert rmed
+                for dst_name in set(output_repldict.keys()):
+                    rmed = node.remove_out_connector(dst_name)
+                    assert rmed
+                for src_name in set(input_repldict.values()):
+                    added = node.add_in_connector(src_name, force=True)
+                    assert added
+                for src_name in set(output_repldict.values()):
+                    added = node.add_out_connector(src_name, force=True)
+                    assert added
+
+                # Update edges
+                for in_edge in state.in_edges(node):
+                    if in_edge.dst_conn in input_repldict:
+                        state.remove_edge(in_edge)
```

Comment:
My recommendation is to not remove the edges but to change the edge fields, because of the nondeterminism/test changes

Reply:
Yes, I had that issue in some tests. That's a good idea.
I will do what you have described.